projects/js-packages/publicize-components/src/types.d.ts

```diff
@@ -0,0 +1,8 @@
+import { SocialScriptData } from './types/types';
+
+// Use module augmentation to add the social property to JetpackInitialState
+declare module '@automattic/jetpack-script-data' {
+	interface JetpackScriptData {
+		social: SocialScriptData;
```

Comment:
It's for TypeScript auto-completions and static type checking.

Reply:
Because `social` is added to `window.JetpackScriptData` via a hook that you can see in `packages/publicize/src/class-publicize-script-data.php` below.

We only inform TypeScript here what we have done from the backend.

Reply:
Thanks for explaining!